srcjs/shinyapp.js

```diff
@@ -631,6 +631,91 @@ var ShinyApp = function() {
     window.location.reload();
   });
 
+  // Helper function for addMessageHandler('shiny-insert-ui').
+  // Turns out that Firefox does not support insertAdjacentElement().
+  // So we have to implement our own version for insertUI.
+  // Code adapted from here: http://forums.mozillazine.org/viewtopic.php?t=445587
+  HTMLElement.prototype.insertAdjacentElement = function(where, parsedNode) {
```

Comment:
I'd suggest making this a standalone function in `utils.js`, instead of modifying `HTMLElement.prototype`.